Assets/MRTK/Services/CameraSystem/ProjectionOverride.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using UnityEngine;
+
+#if WINDOWS_UWP
+using Windows.Perception.Spatial;
+using Windows.Graphics.Holographic;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.CameraSystem
+{
+    /// <summary>
+    /// Helper class to override the projection parameters of the HoloLens frame being presented
+    /// so that smaller details may appear more sharp.  The FOV of the HoloLens will be smaller
+    /// as a tradeoff.
+    /// </summary>
+    /// <remarks>
+    /// Instances of this class are created dynamically by <see cref="MixedRealityCameraSystem"/>.
+    /// So there is no need for an AddComponentMenu attribute.
+    /// </remarks>
+    internal class ProjectionOverride : MonoBehaviour
+    {
+        /// <summary>
+        /// When this is true, projection will be overridden on each frame
+        /// </summary>
+        public bool EnableOverride { get; set; } = false;
+
+#if WINDOWS_UWP
+        private struct InternalHoloLensFrameStructure
+        {
+#pragma warning disable 0649
+            public uint VersionNumber;
+            public uint MaxNumberOfCameras;
+            public IntPtr ISpatialCoordinateSystemPtr;
+            public IntPtr IHolographicFramePtr;
+            public IntPtr IHolographicCameraPtr;
+#pragma warning restore 0649
+        }
+#endif
+
+        /// <summary>
+        /// Coroutine function to set the camera matrices back to their defaults
+        /// </summary>
+        /// <returns>Coroutine enumerator</returns>
+        IEnumerator ResetViewMatricesOnFrameEnd()
+        {
+            yield return new WaitForEndOfFrame();
+            Camera cam = GetComponent<Camera>();
+            cam.ResetStereoViewMatrices();
+            cam.ResetStereoProjectionMatrices();
+        }
+
+        /// <inheritdoc />
+        void OnPreCull()
+        {
+            if (!EnableOverride)
+                return;
+
+            StartCoroutine(ResetViewMatricesOnFrameEnd());
+
+#if WINDOWS_UWP
```

Comment:
Can we include more of this function (or all of it) within the #if WINDOWS_UWP? That way when deployed to other platforms, there won't be another per-frame function call which basically no-ops.

Reply:
Put most of the code into the #if